### PR TITLE
Fix/example codesandbox

### LINKED ISSUE #4988: Broken code sandbox chart examples
A quick look at the chart examples will show that most code sandbox examples are broken. For example, under "Tooltip chart", only the "Wrapped chart" and "Legend" examples are working.

Screen Shot 2020-10-09 at 2 09 16 PM

### PR DESCRIPTION
**Pull request by redallen**
**What**: Remove redundant example names. Parse examples with `acorn` to prepend the title names to an ArrowFunctionExpression.

For instance this Avatar example:
```jsx
import React from 'react';
import { Avatar } from '@patternfly/react-core';
import avatarImg from './examples/avatarImg.svg';

<Avatar src={avatarImg} alt="avatar" />
```

becomes:
```jsx
import React from 'react';
import { Avatar } from '@patternfly/react-core';
import avatarImg from './examples/avatarImg.svg';

Basic = () => <Avatar src={avatarImg} alt="avatar" />
```
based on the `## Basic` example title. These named ArrowFunctionExpressions can easily have `const` prepended to them to be rendered in codesandbox via:
```jsx
const Basic = () => <Avatar src={avatarImg} alt="avatar" />
ReactDOM.render(<Basic />, document.getElementById('root'))
```

The theme now verifies that examples follow one of 3 patterns:
1. Class: `class Example { ... }`
2. Plain JSX: `<Avatar />`
3. Arrow function with block JSX: `Example = () => { ... }` BUT not `Example = () => ()`

Per-md file warning messages have been improved as well:
```sh
./packages/react-charts/src/components/Sparkline/examples/sparkline.md
  28:1-59:4  warning  Example "Basic" must be a class, named block statement, or plain JSX
  62:1-97:4  warning  Example "Green" must be a class, named block statement, or plain JSX

⚠ 2 warnings
./packages/react-datetime/src/components/DatePicker/examples/DatePicker.md
  1:1  warning  Prop component DatePickerLocales missing from tsDocgen

⚠ 1 warning
```

Closes #4988
Closes patternfly/patternfly-org#1791

**Additional issues**:

### CODE REVIEW

**Review by nicolethoen — COMMENTED** [review]
I don't love how the resulting JSX in the docs is formatted, just because it's a little tough to read sometimes, or just makes the formatting seems a little awkward.

I don't know if that's something that can be adjusted?

Screen Shot 2020-10-27 at 1 51 09 PM

File: packages/react-core/src/components/DataList/examples/DataList.md [hunk 1]
```diff
@@ -18,22 +18,6 @@ propComponents:
...
-DataListCheck,
-DataListAction,
-DataListToggle,
-DataListContent,
-Dropdown,
-DropdownPosition,
-KebabToggle,
-DropdownItem,
-} from '@patternfly/react-core';
 import CodeBranchIcon from '@patternfly/react-icons/dist/js/icons/code-branch-icon';
```
Comment (nicolethoen) [inline packages/react-core/src/components/DataList/examples/DataList.md:21]: Is there a reason you keep these imports from react-icons before the examples, but remove the react-core imports?

**Comment by redallen** [comment]
@nicolethoen I thought about adjusting it but decided against it for the time being since to do it properly it required reformatting the user's code. In the case of `<Tooltip>{...}</Tooltip>;` I'd have to move the semicolon so it'd look like:
```
SomeExample = () => (
  <Tooltip>
    {...}
  </Tooltip>
);
```

I'll spend some time tomorrow to see if I can use some AST information to find if the JSX should be multiline and if so use some string manipulation to format it.

File: packages/react-core/src/components/DataList/examples/DataList.md (hunk 1 above)
Comment (redallen) [reply packages/react-core/src/components/DataList/examples/DataList.md:21]: All react-core and react-table imports are included in each react-live instance. There's no need to import them twice.

**Review by kmcfaul — APPROVED** [review]

**Review by dlabrecq — APPROVED** [review]
LGTM. Walked thru each chart example

**Comment by redallen** [comment]
@nicolethoen These examples are now wrapped in `()`:
```jsx
AriaDisabledButtonWithTooltip = () => (
  <Tooltip content="Aria-disabled buttons are like disabled buttons, but focusable. Allows for tooltip support.">
    <Button isAriaDisabled variant="secondary">
      Secondary button to core docs
    </Button>
  </Tooltip>
)
```

**Comment by evwilkin** [comment]
Towards https://github.com/patternfly/patternfly-org/issues/1791

**Review by nicolethoen — APPROVED** [review]
gorgeous

**Review by nicolethoen — CHANGES_REQUESTED** [review]
Wait.... I might have to take that approval back - doesn't your latest update to add the parentheses now break sandbox? isn't the [Title] = () => () pattern what breaks sandbox? When I randomly chose to open the sandbox for the `Button` `Types` sample in code sandbox I get an error...

**Comment by redallen** [comment]
@nicolethoen Agh, the change I made to do the formatting must've broke something. I'll fix it and bump the theme again tomorrow

**Review by kmcfaul — APPROVED** [review]
lgtm going through the component docs

**Review by nicolethoen — APPROVED** [review]

**Review by dlabrecq — COMMENTED** [review]
LGTM

**Pull request merged**